add Microsoft.Extensions.Configuration provider for azure queue streaming

tskimmett opened the pull request:
I'm interested in using Aspire with an Orleans project I'm working on which uses Azure storage for persistence/streaming, but I saw Orleans doesn't yet support the streaming config sent over by the Aspire apphost. I based my changes off of #8764 and did some pretty minimal manual testing using the orleans sample in the Aspire playground.

tskimmett commented on src/Azure/Orleans.Streaming.AzureStorage/Hosting/AzureQueueStreamProviderBuilder.cs:
```diff
@@ -0,0 +1,60 @@
+{
+    public void Configure(ISiloBuilder builder, string name, IConfigurationSection configurationSection)
+    {
+        builder.AddAzureQueueStreams(name, (OptionsBuilder<AzureQueueOptions> optionsBuilder) =>
```
I wasn't sure if this also needs to support configuring the `MessageVisibilityTimeout`. I can get that added in if so.

ReubenBond commented:
@benjaminpetit PTAL - what should IConfiguration support look like for Azure Queues

@tskimmett a test would be helpful here. Eg, update some/all of the existing Azure Queue Streaming tests to use this functionality.

tskimmett commented:
I looked to see if there were relevant tests that could be updated for this and couldn't really tell, but I can take another look at that.

tskimmett commented:
I've updated `AQStreamingTests.cs` to make use of the new configuration approach for streaming. Notably, this uncovered the problem that the ProviderBuilder I added was incomplete, in that it was not registered for the client, only the Silo.

tskimmett commented:
As a side note, I ran into some trouble running tests to begin with for a few reasons:
1. I was missing a basic config `json` file with a connection string
  - I added one on my local machine to get around this. Maybe there are docs explaining local test setup, but I couldn't find any.
2. The tests don't work by default on macOS because the testing framework code is written against storage emulator (deprecated).
  - I got around this by adding a config setting which disables the automatic emulator startup and manually made sure I had Azurite running.

If you'd like, I can open a PR for that last point to slightly improve the dev experience on non-windows platforms, but not sure if you're too concerned about that.

tskimmett commented:
@ReubenBond @benjaminpetit Let me know if you have any more suggestions or feel more tests are necessary.

benjaminpetit commented:
I think it's a good start, but it should also be possible to configure `MessageVisibilityTimeout`.

For tests, I think it would be better to have simple unit tests that just test the config parsing (I believe there is property to get the `AccountName` on the `QueueServiceClient` for example).

tskimmett commented:
Sorry it's been so long, life got in the way, but I am determined to help improve the Aspire integration (albeit selfishly 🙂). Please let me know what else needs to be done to get this over the finish line.

tskimmett commented:
@benjaminpetit Any chance you can take another look?

abbottdev commented:
any updates on this PR @benjaminpetit / @ReubenBond / @jsedlack?
